tests/index.test.ts

```diff
@@ -1,28 +1,117 @@
 import * as cjk from '../src/index.js'
 import { test, expect } from 'vitest'
 
-const testCases: {
-  [char: string]: 'non-cjk' | 'cjk-letter' | 'cjk-punctuation'
-} = {
-  '.': 'non-cjk',
-  'a': 'non-cjk',
-  '。': 'cjk-punctuation',
-  '⿱': 'cjk-punctuation',
-  '中': 'cjk-letter',
-  'ㄅ': 'cjk-letter',
-  '𬉼': 'cjk-letter',
```

Comment:
"𬉼" (in CJK Unified Ideographs ExtensionE), which is previously matched as 'cjk-letter', are matched as 'non-cjk', based on the newest master branch, tested on macOS and GitHub code space (ubuntu).

Reply:
It seems we encountered the UTF-16 issue where [surrogate pair](https://en.wikipedia.org/wiki/UTF-16#Code_points_from_U.2B010000_to_U.2B10FFFF) is split accidentally, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#description) we could probably use `.split(/(?:)/u)` instead of `.split('')` and replace `.charCodeAt(0)` with `.codePointAt(0)` to achieve the goal, and our underlying library `regexp-util` supports the `u` flag via `.toRegExp('u')` as well.

Reply:
Oh, I forgot the surrogate issue of JavaScript. Tests fixed in [6ac3571](https://github.com/ikatyang/cjk-regex/pull/178/commits/6ac3571dbf60de1caa615f65bdebc2f6eecdd1f3).